File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        # Brute Force Approach
        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i] + nums[j] == target:
        #             return [i, j]

        # hashmap
        hashmap = dict()

        for index, value in enumerate(nums):
            if (target - value) in hashmap:
                return [index, hashmap[target - value]]
            
            hashmap[value] = index

        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

    // Brute force approach
    // let n = nums.length;
    // let indices = [];
    // for (let i=0;i<n;i++){
    //     for (let j=i+1;j<n;j++){
    //         if (nums[i] + nums[j] == target){
    //             indices.push(i);
    //             indices.push(j);
    //         }
    //     }
    // }

    // return indices

    // hashmap (value -> index)
    let hashmap = new Map();

    for (let i = 0; i< nums.length; i++){
        let diff = target - nums[i]

        if (hashmap.has(diff)){
           return [hashmap.get(diff), i]
        }

       hashmap.set(nums[i], i)
    }
};